apps/docs/components/Utils/ThemeProvider/ThemeProvider.tsx

```diff
@@ -0,0 +1,39 @@
+"use client";
+
+import React from "react";
+import { getInitialColorMode } from "./ThemeHelper";
+
+export type ColorScheme = "light" | "dark";
+
+interface ColorSchemeContextType {
+    colorMode: ColorScheme;
+    setColorMode?: (newColorScheme: ColorScheme) => void;
```

Comment:
i would make setColorMode required nad change the default value to 

```    
setColorMode: () => {}
```

Reply:
it breaks when I do so

Reply:
whats the error? eslint? you can disable the rule for that line

Reply:
> whats the error

Don't use `{}` as a type. `{}` actually means "any non-nullish value".
- If you want a type meaning "any object", you probably want `object` instead.
- If you want a type meaning "any value", you probably want `unknown` instead.
- If you want a type meaning "empty object", you probably want `Record<string, never>` instead.